notebook/static/notebook/js/notebook.js

```diff
@@ -2844,7 +2844,82 @@ define([
             this._checkpoint_after_save = false;
         }
     };
-    
+
+    Notebook.prototype.save_notebook_as = function() {
+        // If current notebook has some changes, save them, or the copied notebook won't have them.
+        if (this.writable && this.dirty) {
+            this.save_notebook();
+        }
+        var that = this;
+        var dialog_body = $('<div/>').append(
+            $("<p/>").addClass("save-message")
+                .text(i18n.msg._('Enter a notebook path relative to notebook dir'))
+        ).append(
+            $("<br/>")
+        ).append(
+            $('<input/>').attr('type','text').attr('size','25')
+            .addClass('form-control').attr('placeholder', that.notebook_name)
+        );
+
+        var d = dialog.modal({
+            title: 'Save As',
+            body: dialog_body,
+            keyboard_manager: this.keyboard_manager,
+            notebook: this,
+            buttons: {
+                Cancel: {},
+                Save: {
+                    class: 'btn-primary',
+                    click: function() {
+                        var nb_path = d.find('input').val();
+                        var nb_name = nb_path.split('/').slice(-1).pop();
+                        // If notebook name does not contain extension '.ipynb' add it
+                        var ext = utils.splitext(nb_name)[1];
+                        if (ext === '') {
+                            nb_name = nb_name + '.ipynb';
+                            nb_path = nb_path + '.ipynb';
+                        }
+                        var save_thunk = function() {
+                            var model = {
+                                'type': 'notebook',
+                                'content': that.toJSON(),
+                                'name': nb_name
+                             };
+                            return that.contents.save(nb_path, model)
+                                .then(function(data) {
+                                    window.open(data.path, '_self');
```

Comment:
Here I think we should use the same functions that rename uses to update the session and the page:

https://github.com/jupyter/notebook/blob/2aca6f98b0e46c9a3d635faf0c03abf113ff4089/notebook/static/notebook/js/notebook.js#L2984-L2985